### PR TITLE
Upvote/Downvote 

### PR DESCRIPTION
**Pull request by zaverichintan**
added like dislike functionality. User can click as many time user wishes.

TBD - 
1 Like per user

### CODE REVIEW

**Comment by pmuens** [comment]
Cool! A check if someone has already up / downvoted would be great!

**Comment by mattvella07** [comment]
Yea I agree, 1 like per user.  And a way to remove their up/down vote would be good too.  Maybe by clicking the up/down vote icon a second time will remove their up/down vote, like a toggle.

**Comment by zaverichintan** [comment]
Ok sure.

**Comment by zaverichintan** [comment]
I am trying something like this -
`
 if(Posts.find({ already_voted:  Meteor.userId() }) ){
 Posts.update( { _id: _id }, { $pull: { already_voted: Meteor.userId() } });
   console.log(" found");

 }else {
    Posts.update( { _id: _id }, { $push: { already_voted: Meteor.userId() } });
   console.log("not found");
 }
`

**Comment by mattvella07** [comment]
Is already_voted an array containing userIDs of all users who voted on a particular post?

It's hard to be sure without seeing all the code, but I think I see the issue.  When you are doing this part:
`Posts.find({ already_voted: Meteor.userId() })`

That is looking for any post that has already_voted with a value of the current userId, so you need to include the id of the post in your find as well.  Something like this: 
`Posts.find({ _id: post_id, already_voted: Meteor.userId() })`

But it doesn't look like the post id is stored in the id field of the DIV, so you would have to add that first that way you can get the post_id of the post that the user is trying to up/down vote.  Hopefully this all makes sense.

**Comment by zaverichintan** [comment]
I agree to your approach. Tried doing that. The thing is that when song for the first time, field does not exist and it should give not found, instead it gives found.

**Comment by mattvella07** [comment]
Hmmm...why don't you try checking if the count > 0 instead: 
`if(Posts.find({ already_voted:  Meteor.userId() }).count() > 0 )`

**Comment by zaverichintan** [comment]
I will surely give it a try.  Thank you.

**Comment by zaverichintan** [comment]
I want to check if user id exists inside an array field of mongodb (using meteor)

db.posts.find().pretty()

 {
        "_id" : "hT3ezqEyTaiihoh6Z",
        "body" : "hey\n",
        "authorId" : "AyJo5nf2Lkdqd6aRh",
        "createdAt" : ISODate("2016-05-13T06:19:34.726Z"),
        "updatedAt" : ISODate("2016-05-13T06:19:34.726Z"),
        "likecount" : 0,
        "already_voted" : [ ] }

 db.posts.find( { _id:"hT3ezqEyTaiihoh6Z"},{ already_voted: { $in : ["AyJo5nf2Lkdqd6aRh"]} }).count()

 1
It gives count value 1 , where as I am expecting it to be 0 .

**Comment by zaverichintan** [comment]
The color of thumb does not change. Is there some issue with the helper part ?

**Comment by mattvella07** [comment]
I would say if it is liked add a class to the button and you can have different styles in the LESS file based on button.className

File: client/views/shared/_post.html [hunk 1]
```diff
@@ -12,8 +12,11 @@
           <span class="fa {{#if author.status.online}} fa-circle presence online {{else}} fa-circle-o presence offline {{/if}} "></span>
           {{#if belongsPostToUser}}
             <button class="u-pull-right no-button remove-post" data-id="remove-post"><i class="fa fa-trash"></i></button>
+
           {{/if}}
           <p class="u-pull-right formatted-date">{{formatDate createdAt}}</p>
+
+          <button class="no-button " data-id="like-post"><i class="fa fa-thumbs-o-up {{#if isLiked}}liked{{/if}}" aria-hidden="false"></i>{{likecount}}</button>
```
Comment (zaverichintan) [inline client/views/shared/_post.html]: Here I am trying to add class.
Is the helper function OK?

**Comment by mattvella07** [comment]
I don't think `self._id` is giving you the id of the post like you are expecting, but I can't be sure.  So try to console.log() it right before the Meteor.call to verify that it is giving you the correct post id.

**Comment by zaverichintan** [comment]
self._id is giving the id of post, the issue is that the function gets called two times.

**Comment by mattvella07** [comment]
It looks like those are all different IDs.  Are you referring to the isLiked function that gets called twice?  Because that shouldn't matter it should return the same value both times.  Or are you referring to the click event for like that gets called twice?  Because that would be an issue since it would like it then unlike it.

**Comment by zaverichintan** [comment]
The 1st  and 4th are same as there are 3 posts. Isliked function in called twice ss meteor call makes 2 times calls.

**Comment by mattvella07** [comment]
I see.  If you want you can commit what you have to your fork of gravity and I can download it and try to see if I can figure out a way to fix it.

**Comment by zaverichintan** [comment]
Done with the updated commit. 
Toggle of likes is working. 
The color when class liked is added changes to blue is done. 
The helper function perform same function (meteor call)  two times.

**Comment by pmuens** [comment]
@zaverichintan Alright. This means that it's ready to be reviewed?

**Comment by mattvella07** [comment]
The toggling if likes is working, but I'm not seeing the color change after a post has been liked.  And the name and like button are lined up, not a huge deal but would be nice to have it all lined up.

**Comment by zaverichintan** [comment]
The toggling part works but to implement  color change If already liked is not working due to 2 calls to the function. I am not sure why that thing happens may be due server and client side rendering.

**Comment by mattvella07** [comment]
@zaverichintan Ok, I think I figured it out.  Most of what you have is correct, but the issue is that Meteor.call is asynchronous so the view is rendered before the callback function executes which is why the 'liked' class is never assigned.  If you remove the Meteor.call and just do the Posts.find() in the isLiked helper function it will work.  

@pmuens If you want to merge this PR I can make the adjustments or @zaverichintan you can make them if you like.

**Comment by zaverichintan** [comment]
I will try it out. The async behavior of meteor call us the reason behind multiple calls that's the reason in my opinion too.
I Will try to have direct access to collection.

**Comment by mattvella07** [comment]
Here's what I had: 

```
isLiked: function() {
    if(Posts.find( { _id: this._id , already_voted: { "$in" : [Meteor.userId()]} }).count() === 1) {
      return 'liked';
    } 
      
    return '';
  }
```

**Comment by zaverichintan** [comment]
Done. Thanks a lot @mattvella07 . Meteor call occurs two times, so subscribing the data (posts) and then rendering directly in helper.

**Comment by pmuens** [comment]
Good job! Thank you!

**Pull request merged**